File: alarm-handler-archetype/src/main/resources/archetype-resources/src/main/java/__pluginName__/__pluginName__AlarmHandlerDescriptor.java
package ${package}.${pluginName};

import se.curity.identityserver.sdk.plugin.descriptor.AlarmHandlerPluginDescriptor;
import se.curity.identityserver.sdk.alarm.AlarmHandler;

public final class ${pluginName}AlarmHandlerDescriptor implements AlarmHandlerPluginDescriptor<${pluginName}AlarmHandlerConfig>
{
    @Override
    public Class<? extends AlarmHandler> getAlarmHandler() {
        return ${pluginName}AlarmHandler.class;
    }

    @Override
    public String getPluginImplementationType()
    {
        return "${pluginName.toLowerCase()}";
    }

    @Override
    public Class<? extends ${pluginName}AlarmHandlerConfig> getConfigurationType()
    {
        return ${pluginName}AlarmHandlerConfig.class;
    }    
}

File: kotlin-alarm-handler-archetype/src/main/resources/archetype-resources/src/main/kotlin/__pluginName__/__pluginName__AlarmHandlerDescriptor.kt
package ${package}.${pluginName};

import se.curity.identityserver.sdk.plugin.descriptor.AlarmHandlerPluginDescriptor;
import se.curity.identityserver.sdk.alarm.AlarmHandler;

class ${pluginName}AlarmHandlerDescriptor: AlarmHandlerPluginDescriptor<${pluginName}AlarmHandlerConfig>
{
    override fun getAlarmHandler() = ${pluginName}AlarmHandler::class.java

    override fun getPluginImplementationType() = "${pluginName.toLowerCase()}"

    override fun getConfigurationType() = ${pluginName}AlarmHandlerConfig::class.java;
}
